src/t_list.c

```diff
@@ -363,26 +363,30 @@ void lsetCommand(client *c) {
     }
 }
 
+void listElementsRemoved(client *c, int where, robj *o, robj *value) {
+    char *event = (where == LIST_HEAD) ? "lpop" : "rpop";
+
+    decrRefCount(value);
```

Comment:
in my eyes, the decrRefcount is not part of the responsibility of this function.
it shouldn't deal with the value.
it didn't removed it from the list, and it didn't put it in the reply buffer, so it doesn't suppose to know if the refcount needs decrement or not, that's part of the responsibilities of the caller.
this function is just about what needs to be done the key, keyspace and general server tasks after the list was modified.